analyzers/src/SonarAnalyzer.CFG/ShimLayer/AnalysisContext/SymbolStartAnalysisContext.cs

```diff
@@ -0,0 +1,169 @@
+        cancellationTokenAccessor = CreatePropertyAccessor<CancellationToken>(symbolStartAnalysisContextType, nameof(CancellationToken));
+        compilationAccessor = CreatePropertyAccessor<Compilation>(symbolStartAnalysisContextType, nameof(Compilation));
+        optionsAccessor = CreatePropertyAccessor<AnalyzerOptions>(symbolStartAnalysisContextType, nameof(Options));
+        symbolAccessor = CreatePropertyAccessor<ISymbol>(symbolStartAnalysisContextType, nameof(Symbol));
+        registerCodeBlockAction = CreateRegistrationMethod<CodeBlockAnalysisContext>(symbolStartAnalysisContextType, nameof(RegisterCodeBlockAction));
+        registerCodeBlockStartActionCS =
+            CreateRegistrationMethod<CodeBlockStartAnalysisContext<CS.SyntaxKind>>(symbolStartAnalysisContextType, nameof(RegisterCodeBlockStartAction), typeof(CS.SyntaxKind));
+        registerOperationAction =
+            CreateRegistrationMethodWithAdditionalParameter<OperationAnalysisContext, ImmutableArray<OperationKind>>(symbolStartAnalysisContextType, nameof(RegisterOperationAction));
+        registerOperationBlockAction = CreateRegistrationMethod<OperationBlockAnalysisContext>(symbolStartAnalysisContextType, nameof(RegisterOperationBlockAction));
+        registerOperationBlockStartAction = CreateRegistrationMethod<OperationBlockStartAnalysisContext>(symbolStartAnalysisContextType, nameof(RegisterOperationBlockStartAction));
+        registerSymbolEndAction = CreateRegistrationMethod<SymbolAnalysisContext>(symbolStartAnalysisContextType, nameof(RegisterSymbolEndAction));
+        registerSyntaxNodeActionCS = CreateRegistrationMethodWithAdditionalParameter<SyntaxNodeAnalysisContext, ImmutableArray<CS.SyntaxKind>>(
+            symbolStartAnalysisContextType, nameof(RegisterSyntaxNodeAction), typeof(CS.SyntaxKind));
+    }
+
+    public SymbolStartAnalysisContext(object roslynSymbolStartAnalysisContext)
+    {
+        RoslynSymbolStartAnalysisContext = roslynSymbolStartAnalysisContext;
+    }
+    public object RoslynSymbolStartAnalysisContext { get; }
+    public CancellationToken CancellationToken => cancellationTokenAccessor(RoslynSymbolStartAnalysisContext);
+    public Compilation Compilation => compilationAccessor(RoslynSymbolStartAnalysisContext);
+    public AnalyzerOptions Options => optionsAccessor(RoslynSymbolStartAnalysisContext);
+    public ISymbol Symbol => symbolAccessor(RoslynSymbolStartAnalysisContext);
+
+    public void RegisterCodeBlockAction(Action<CodeBlockAnalysisContext> action) =>
+        registerCodeBlockAction(RoslynSymbolStartAnalysisContext, action);
+
+    public void RegisterCodeBlockStartAction<TLanguageKindEnum>(Action<CodeBlockStartAnalysisContext<TLanguageKindEnum>> action) where TLanguageKindEnum : struct
+    {
+        var languageKindType = typeof(TLanguageKindEnum);
+        if (languageKindType == typeof(CS.SyntaxKind))
+        {
+            var casted = (Action<CodeBlockStartAnalysisContext<CS.SyntaxKind>>)action;
```

Comment:
`cast` is invariable in English. Besides, I don't think we need to define a variable. In this case, we can inline:
```cs
registerCodeBlockStartActionCS(roslynSymbolStartAnalysisContext, (Action<CodeBlockStartAnalysisContext<CS.SyntaxKind>>)action);
```